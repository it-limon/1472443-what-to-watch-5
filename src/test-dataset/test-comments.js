export const testComments = [
  {
    id: 1,
    user: {
      id: 1,
      name: `User 1`
    },
    rating: 8,
    comment: `Comment 1`,
    date: `1995-12-17T03:24:00`
  },
  {
    id: 2,
    user: {
      id: 2,
      name: `User 2`
    },
    rating: 7,
    comment: `Comment 2`,
    date: `1996-12-17T03:24:00`
  },
  {
    id: 3,
    user: {
      id: 3,
      name: `User 3`
    },
    rating: 5,
    comment: `Comment 3`,
    date: `1997-12-17T03:24:00`
  }
];