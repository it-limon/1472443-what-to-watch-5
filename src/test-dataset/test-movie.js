export const testMovie = {
  id: 1,
  name: `El Camino: A Breaking Bad Movie`,
  posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
  previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
  backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
  backgroundColor: `#000000`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
  rating: 7.2,
  scoresCount: 36346,
  director: `Vince Gilligan`,
  starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
  runTime: 122,
  genre: `Drama`,
  released: 2019,
  isFavorite: true
};
