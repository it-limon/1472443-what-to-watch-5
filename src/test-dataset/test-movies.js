export const testMovies = [
  {
    id: 1,
    name: `El Camino: A Breaking Bad Movie`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `Fugitive Jesse Pinkman runs from his captors, the law, and his past.`,
    rating: 7.2,
    scoresCount: 36346,
    director: `Vince Gilligan`,
    starring: [`Aaron Paul`, `Jonathan Banks`, `Matt Jones`],
    runTime: 122,
    genre: `Drama`,
    released: 2019,
    isFavorite: false
  },
  {
    id: 2,
    name: `The Dark Knight`,
    posterImage: `https://cdn.pixabay.com/photo/2020/09/27/04/15/cat-5605615_960_720.jpg`,
    previewImage: `https://cdn.pixabay.com/photo/2020/11/16/16/13/trees-5749406_960_720.jpg`,
    backgroundImage: `https://cdn.pixabay.com/photo/2020/11/16/16/05/hoverfly-5749361_960_720.jpg`,
    backgroundColor: `#000000`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    description: `When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, 
                  Batman must accept one of the greatest psychological and physical tests of his ability 
                  to fight injustice.`,
    rating: 9,
    scoresCount: 2271245,
    director: `Christopher Nolan`,
    starring: [`Christian Bale`, `Heath Ledger`, `Aaron Eckhart`],
    runTime: 152,
    genre: `Crime`,
    released: 2008,
    isFavorite: true
  }
];