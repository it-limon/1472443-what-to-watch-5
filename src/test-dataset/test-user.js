export const testUser = {
  id: 1,
  email: `user@mail.ru`,
  name: `User Name`,
  avatarUrl: `https://cdn.pixabay.com/photo/2017/06/23/08/13/china-2433891_960_720.png`
};