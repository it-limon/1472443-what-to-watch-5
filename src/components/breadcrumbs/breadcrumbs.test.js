import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Breadcrumbs from "./breadcrumbs";
import {Provider} from "react-redux";
import {Route, BrowserRouter} from "react-router-dom";
import {testMovie} from "../../test-dataset/test-movie";

describe(`Render Breadcrumbs`, () => {
  const mockStore = configureStore([]);
  let store = null;
  let breadcrumbsComponent = null;

  beforeEach(() => {
    store = mockStore({
      STATE: {
        lastActiveMovie: testMovie
      }
    });

    breadcrumbsComponent = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <Breadcrumbs />
            </Route>
          </BrowserRouter>
        </Provider>
    );
  });

  it(`Should Breadcrumbs to store render correctly`, () => {
    expect(breadcrumbsComponent.toJSON()).toMatchSnapshot();
  });

});
