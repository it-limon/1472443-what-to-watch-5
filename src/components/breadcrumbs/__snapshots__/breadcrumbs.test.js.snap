// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Breadcrumbs Should Breadcrumbs to store render correctly 1`] = `
<nav
  className="breadcrumbs"
>
  <ul
    className="breadcrumbs__list"
  >
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
        href="/films/1"
        onClick={[Function]}
      >
        El Camino: A Breaking Bad Movie
      </a>
    </li>
    <li
      className="breadcrumbs__item"
    >
      <a
        className="breadcrumbs__link"
      >
        Add review
      </a>
    </li>
  </ul>
</nav>
`;
